packages/core/admin/ee/admin/hooks/useLicenseLimitNotification/index.js

```diff
@@ -4,110 +4,76 @@
-      title: {
-        id: 'notification.ee.warning.over-seat-limit.title',
-        defaultMessage: 'Over seat limit ({currentUserCount}/{permittedSeats})',
-        values: { currentUserCount, permittedSeats },
-      },
-      link: {
-        url: linkURL,
-        label: linkLabel,
-      },
-      blockTransition: true,
-    };
-
-    return notification;
-  }
-
-  if (licenseLimitStatus === 'AT_LIMIT') {
-    notification = {
-      type: 'softWarning',
-      message: {
-        id: 'notification.ee.warning.at-seat-limit',
-        defaultMessage:
-          "Add seats to re-enable users. If you already did it but it's not reflected in Strapi yet, make sure to restart your app.",
-      },
-      title: {
-        id: 'notification.ee.warning.at-seat-limit.title',
-        defaultMessage: 'At seat limit ({currentUserCount}/{permittedSeats})',
-        values: { currentUserCount, permittedSeats },
-      },
-      link: {
-        url: linkURL,
-        label: linkLabel,
-      },
-      blockTransition: true,
-    };
-
-    return notification;
-  }
-
-  return notification;
-};
-
-const shouldDisplayNotification = (pathname) => {
-  const isLocation = (string) => pathname.includes(string);
-  const shownInSession = window.sessionStorage.getItem(`notification-${pathname}`);
-
-  if (isLocation('/') && shownInSession) {
-    return false;
-  }
-
-  if (isLocation('users') && shownInSession) {
-    return false;
-  }
-
-  return true;
-};
+const BILLING_STRAPI_CLOUD_URL = 'https://cloud.strapi.io/profile/billing';
+const BILLING_SELF_HOSTED_URL =
+  'https://strapi.chargebeeportal.com/portal/v2/login?forward=portal_main';
 
 const useLicenseLimitNotification = () => {
-  let licenseLimitInfos = useLicenseLimitInfos();
+  const { formatMessage } = useIntl();
+  let { license } = useLicenseLimits();
   const toggleNotification = useNotification();
-  const location = useLocation();
+  const { pathname } = useLocation();
 
-  // eslint-disable-next-line consistent-return
   useEffect(() => {
-    if (!licenseLimitInfos || !licenseLimitInfos.permittedSeats) return;
-
-    if (!shouldDisplayNotification(location.pathname)) return;
+    if (!license?.data) {
+      return;
+    }
 
     const { currentUserCount, permittedSeats, licenseLimitStatus, isHostedOnStrapiCloud } =
-      licenseLimitInfos;
-    const notification = notificationBody(
-      currentUserCount,
-      permittedSeats,
-      licenseLimitStatus,
-      isHostedOnStrapiCloud
-    );
-    const onClose = () => window.sessionStorage.setItem(`notification-${location.pathname}`, true);
-    toggleNotification({ ...notification, onClose });
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+      license?.data ?? {};
+
+    const shouldDisplayNotification =
+      permittedSeats &&
+      window.sessionStorage.getItem(`${STORAGE_KEY_PREFIX}-${pathname}`) &&
+      ['/', '/users'].every(pathname.includes);
```

Comment:
@ivanThePleasant I think instead of doing this it would be better to get rid of the condition and to call `useLicenseLimitNotification()` on both pages.

Reply:
Wouldn't then each page need to track if notification was displayed on it already?

Reply:
You could pass the pathname into the hook as argument, so the hook takes care of the storage.

Reply:
I personally don't see if that's doing any major difference, it seems that having all this logic in one place is a bit less complexity. But if you insist/say that it's better for performance or something, the I can change that

Reply:
I think it won't perform better, but: it will scale better. With a central check like that there is a good chance we will keep on adding cases and therefore this piece of code will become more and more convoluted and complicated. If we just spread it out there is less of chance of this to happen I think.

Reply:
I was just checking it and you've actually changed the logic of it so it stopped working as intended. Here's the intended use: 
```
const shouldDisplayNotification =
      permittedSeats && !window.sessionStorage.getItem(`${STORAGE_KEY_PREFIX}-${pathname}`);
```

If the session storage returns true, this means notification has already been shown in this session and therefore `shouldDisplayNotification` is false. And we don't need to check the paths there because this hook is called on specific pages, so it will only show on those pages